### ISSUE TITLE
MessageChat logs spam (BB #524)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#524]
This issue was migrated from bitbucket.
**Original Reporter:** 
**Original Date:** 04.03.2014 05:40:23 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524

<hr>

SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 100)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 90)
04-03-14 06:32
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 92)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 84)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 98)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 90)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 92)
04-03-14 06:32
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 84)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 98)
SESSION: opcode CMSG_MESSAGECHAT (0x0095) has unprocessed tail data (read stop at 8 from 90)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 04.03.2014 23:20:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-8919139

<hr>

Could you please update to changeset 09cce6b ? These data posted are not very useful. If you post this again with the additional data (provided from the changeset), it will be much easier to solve this problem.

This is most likely done by an addon, though, I think.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 05.03.2014 10:26:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-8925738

<hr>

I tested it, and Zaffy´s idea is right. I get same problem if i turned off Addon channel.

AddonChannel = 0 - will be spammed 20rows/sec
AddonChannel = 1 is clean without problems

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 07.04.2014 23:09:22 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-9540609

<hr>

Can this be closed?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 08.04.2014 05:54:48 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-9544333

<hr>

i had same problem, stopping addon channel fixed for me this problem

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.07.2014 03:03:28 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-11119682

<hr>

Is this something that needs to be fixed?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 08.07.2014 06:59:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-11122274

<hr>

I think yes, addons spaming a lot. Spamed logs section, addons are logged with world channels. When stop addon channel in config. They will spam server.log

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 10.07.2014 15:54:35 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-11174761

<hr>

Btw, this is what I did to resolve it for me. Zaffy didn't approve of it though :P

```
#!c++

# HG changeset patch
# User LordUsagi
# Date 1397150429 10800
#      Thu Apr 10 14:20:29 2014 -0300
# Node ID 0450c3f42c23b3a36a3194bd0eb1cf740fc2e6cf
# Parent  54be61ea3d1e395bb3494f799f238146d78f035d
[Core/Chat] Fix issue #524. Unprocessed tail data spam from AddonChannel

diff -r 54be61ea3d1e -r 0450c3f42c23 src/game/ChatHandler.cpp
--- a/src/game/ChatHandler.cpp  Thu Apr 10 13:15:00 2014 -0300
+++ b/src/game/ChatHandler.cpp  Thu Apr 10 14:20:29 2014 -0300
@@ -119,9 +119,12 @@
                 GetPlayer()->GetName(), msg.c_str());
         }

-        // Disabled addon channel?
+        // check if addon messages are disabled
         if (!sWorld.getConfig(CONFIG_ADDON_CHANNEL))
+        {
+            recv_data.rpos();
             return;
+        }
     }
     // LANG_ADDON should not be changed nor be affected by flood control
     else
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 10.07.2014 15:55:22 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-11174779

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 19.08.2014 22:33:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/524/#comment-11877927

<hr>

[Core/Chat] Fix issue #524. Unprocessed tail data spam from AddonChannel

→ <<cset 19e3bdc5635d>>
